doc/algorithm.md

```diff
@@ -0,0 +1,271 @@
+
+## Table of contents
+
+0. [Tiles and tilesets](#tiles-and-tilesets)
+0. [Rearranging source tiles](#rearranging-source-tiles)
+0. [Terrain hyperraph](#terrain-hypergraph)
+0. [Combining terrains](#combining-terrains)
+0. [Generating corner masks](#generating-corner-masks)
+
+## Tiles and tilesets
+
+Wangscape's input is:
+* A set of image files containing your source terrain tiles.
+* A set of configuration files describing the locations of the source terrain tiles and how to combine them into tilesets. The exact format is specified in the [options schema](./schemas/options_schema.json), and includes:
+  * The resolution (in pixels) of tiles used.
+  * The image file and offset within the image of each terrain.
+  * A set of terrain cliques (see [Terrain hypergraph](#terrain-hypergraph)).
+  * Filenames of noise module groups used for [generating corner masks](#generating-corner-masks).
+  * Which method is used to convert noise values into corner masks ([generating corner masks](#generating-corner-masks)).
+* A set of JSON files describing module groups used to [generate noise values](#generating-noise-values).
+
+Wangscape's output is:
+* A set of image files, each one containing a full corner Wang tileset for a subset of the input terrain types,
+all of them with compatible borders (if configured correctly).
+* A set of JSON files describing the tilesets generated, and where to find a specific tile within them.
+
+The schemas defining the JSON formats used by Wangscape are found in the `doc/schemas` directory,
+and can be viewed in human-readable format using [docson](https://wangscape.github.io/docson/) by [@lbovet](https://github.com/lbovet).
+
+### Input schema docson links
+* [Options schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/options_schema.json)
+* [Module group schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/module_group_schema.json)
+
+### Metaoutput schema docson links
+* [Tiles schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/tiles_schema.json)
+* [Tile groups schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/tile_groups_schema.json)
+* [Tilesets schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/tilesets_schema.json)
+* [Terrain hypergraph schema](https://wangscape.github.io/docson/#https://raw.githubusercontent.com/Wangscape/Wangscape/master/doc/schemas/terrain_hypergraph_schema.json)
+
+## Rearranging source tiles
+
+Wangscape's output tiles are designed to be displayed offset by a quarter tile.
+In order for the display tiles to match the source tiles,
+the source images must be chopped into quarters and reassembled in this arrangement:
+```
+12 -> 43
+34    21
+```
```

Comment:
Compare `doc/examples/example3/void.png` to `doc/examples/example3/output/g.v.png` and to Wangview's display of `example3`.

In `void.png` the purple circle has its centre in the middle of the tile. However, in `g.v.png` the circle must have its centre at the corner. That is the effect of this rearrangement. In Wangview, the circles have their centres correctly in the middle of the logical tiles, because the rearranged tiles are drawn with a half-tile offset in both axes.